### DISCUSSION TITLE
Can't install Wix4 RC1

### DISCUSSION THREAD

**Discussion by e-t-l** [wixtoolset/issues#7104] [Questions] [answered]
I installed .NET 7 and ran `dotnet tool install --global wix --version 4.0.0-rc.1` (I did not have wix previously installed)
It failed with the following output: 
```
C:\Users\Etl\AppData\Local\Temp\aa814961-35a8-4543-bef9-923007e39c08\restore.csproj : error NU1101: Unable to fin
d package wix. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
The tool package could not be restored.
Tool 'wix' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool
```

Obviously, I did specify version, since I copied and pasted the command. Same thing re: the name of the tool. I assume there wouldn't be an issue with the package being a .Net tool, since it was designed as such. My internet connection is fine, although I don't know how to check if there's an issue wthi the NuGet feed. 

Running Windows 11 Home

**Comment by robmen** [comment]
Documentation needs to point out that you need to add `--prerelease` until WiX v4 finally reaches `4.0.0`.

**Reply by robmen** [reply]
Actually, @barnson pointed out to me the `--prerelease` switch is NOT necessary. In fact, it is not allowed when you specify `--version`. So, the current documentation _is_ correct and I tried it on my machine and it worked. So we'd need more information to know what your issue might be.

**Reply by e-t-l** [reply]
I'm pretty new to .NET, NuGet, and Wix. What sort of information would be helpful? I'm happy to do as much troubleshooting and information-digging as I can; I appreciate you folks in this community helping me out!

**Comment by barnson** [comment]
> C:\Users\Etl\AppData\Local\Temp\aa814961-35a8-4543-bef9-923007e39c08\restore.csproj : error NU1101: Unable to find package wix. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages

There's no mention of `nuget.org`. Do you have a nuget.config that removes it?

**Reply by e-t-l** [reply]
I do not have any nuget.config files that do anything involving nuget.org.

This is my first time installing Wix (actually, my first time using .NET and Nuget as well - I'm new to all this) so I have no idea what should or should not be mentioned.

**Comment by barnson** [comment]
Try running `nuget sources`. That should list nuget.org as a source.

**Reply by e-t-l** [reply]
`nuget sources` outputs the following: 
```
WARNING: Could not load file or assembly 'HolographicExtensions.dll' or one of its dependencies. The module was expected to contain an assembly manifest.
WARNING: Could not load file or assembly 'HoloSHExtensions.dll' or one of its dependencies. The module was expected to contain an assembly manifest.
WARNING: Could not load file or assembly 'Windows.Devices.Printers.Extensions.dll' or one of its dependencies. The module was expected to contain an assembly manifest.
Registered Sources:
  1.  Microsoft Visual Studio Offline Packages [Enabled]
      C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\
```

(Just to check, running `dotnet nuget version` outputs `NuGet Command Line 6.4.0.117`. Is that as expected?)

**Comment by e-t-l** [comment, marked as answer]
The issue was caused, for some reason, by the `nuget.config` file created by Visual Studio 2019. Following the suggestions from [this NuGet wiki page](https://github.com/NuGet/Home/wiki/nuget.org-not-used-on-new-machine), I simply deleted every file named `nuget.config` on my computer (since I don't really use Visual Studio or NuGet for anything else, I was not worried about this breaking any previous configurations). After that, Wix installed correctly!

Per the NuGet wiki, if you don't want to delete nuget.config, it would likely also work to manually add nuget.org as a source by running:
`dotnet nuget add source https://api.nuget.org/v3/index.json --name nuget.org`
OR
`nuget.exe sources add -Name nuget.org -Source https://api.nuget.org/v3/index.json`
